core/core-configuration-layer.el

```diff
@@ -1953,6 +1939,26 @@ FILE-TO-LOAD is an explicit file to load after the installation."
             (1+ configuration-layer-error-count))
     (setq configuration-layer-error-count 1)))
 
+(defun configuration-layer-check-archive-status (archive &optional recheck)
+  "Check ARCHIVE status.
+
+By default status is checked only when current status of ARCHIVE
+is `unknown'. Check is forced when RECHECK is non-nil."
+  (let* ((obj (assoc archive configuration-layer--elpa-archives-status))
+         (state (cdr obj))
+         (url (cdr (assoc archive package-archives))))
+    (message "archive %s state %s url %s" archive state url)
+    (when (and (or (eq state 'unknown) recheck)
+               url)
+      (condition-case nil
+          (if (url-http-file-exists-p url)
+              (setq state 'avaialble)
+            (setq state 'unavailable))
+        ((error) (setq state 'unavailable)))
+      (setq configuration-layer--elpa-archives-status (delq obj configuration-layer--elpa-archives-status))
+      (add-to-list 'configuration-layer--elpa-archives-status (cons archive state)))
```

Comment:
You can do

    (setcdr (assq archive configuration-layer--elpa-archives-status) state)